wolfcrypt/src/dh.c

```diff
@@ -2152,6 +2150,10 @@ int wc_DhGenerateParams(WC_RNG *rng, int modSz, DhKey *dh)
             ret = MP_INIT_E;
         }
     }
+    else {
+        (void)XMEMSET(&tmp, 0, sizeof(mp_int));
+        (void)XMEMSET(&tmp2, 0, sizeof(mp_int));
+    }
```

Comment:
Could you explain why this is needed? This case only executes if `ret != 0`, in which case `tmp` and `tmp2` will only be accessed at the end of this function by:
```
    mp_clear(&tmp);
    mp_clear(&tmp2);
```

Reply:
`mp_clear` is reading members `dp` and `used`.

Reply:
Hmm. I think it may be better to add a boolean (`tmp_valid`) that gets set when the call to `mp_init_multi` succeeds. Then wrap the calls to `mp_clear` in a conditional based on  `tmp_valid`.